### ISSUE TITLE
[a11y]: ai-chat should allow JAWS to identify interactive controls at message level

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#874]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

Regenerate Button , thumps up /thumps down buttons , All the links in the link list; basically all interactive elements in the chat bot are not identifiable by JAWS.

Issue description: Interactive controls in the Chat window are screen reader /keyboard in accessible :Controls are not identifiable by Screen reader, label, role, state etc. not identified on tab/arrow key navigation.

Recommended fix: Make sure all the Controls /Content are reachable and readable by using Standard keyboard methods along with Screen reader :

Use semantic HTML elements for controls whenever possible as they are natively keyboard accessible.

For custom controls, ensure that the element is focusable using tab index="0" and operable with standard keyboard interactions.

Make sure Descriptive Label Appropriate Role , State are in place for ensuring the Screen reader accessibility (use ARIA /HTML Technics )

User impact: If the chat window is not readable or navigable, the entire chat content becomes inaccessible for screen reader users. This prevents them from reviewing
user queries, AI responses, and interactive elements, resulting in a complete loss of access to the conversation.

Related to #873.

### WCAG 2.1 Violation

WCAG Mapping: 4.1.2 Name, Role, Value

### Reproduction/example

n/a

### Steps to reproduce

1. make sure JAWS is active
2. open a-chat
3. ask a question in ai-chat that returns a link list, e.g. "what documentation should I start with to get familiar with Cloud?"
4. at the response level, use JAWS to interact with any button, e.g. thumbs up, down, regenerate, the link list elements, ...
5. see that the elements are inaccessible by JAWS

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
